src/saltext/vmware/states/roles.py

```diff
@@ -0,0 +1,161 @@
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+
+import saltext.vmware.modules.roles as roles_module
+import saltext.vmware.utils.connect as connect
+import saltext.vmware.utils.drift as drift
+
+log = logging.getLogger(__name__)
+
+try:
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+__virtualname__ = "vcenter_roles"
+__proxyenabled__ = ["vcenter_roles"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def config(name, config, service_instance=None, profile=None):
+    """
+    Get/Set roles configuration based on drift report.
+
+    name
+        Name of configuration. (required).
+
+    config
+        List of objects with configuration values. (required).
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    profile
+        Profile to use (optional)
+
+    .. code-block:: yaml
+
+        vcenter_roles_config_example:
+            vcenter_roles.config:
+                - profile: vcenter
+                - config:
+                - role: SRM Administrator
+                    groups:
+                    - group: SRM Protection
+                        privileges:
+                        - Stop
+                        - Protect
+                    - group: Recovery History
+                        privileges:
+                        - Delete History
+                        - View Deleted Plans
+                    - group: Recovery Plan
+                        privileges:
+                        - Configure commands
+                        - Create
+                        - Remove
+                        - Modify
+                        - Recovery
+                    - group: Protection Group
+                        privileges:
+                        - Assign to plan
+                        - Create
+                        - Modify
+                        - Remove
+                        - Remove from plan
```

Comment:
**suggestion** this formatting is definitely borked - Salt errors with "mapping values are not allowed in this context" when I stick this in an SLS file.